server/proxy/service.go

```diff
@@ -62,20 +67,47 @@ type Service interface {
 
 var _ Service = (*Proxy)(nil)
 
-func NewService(mm *pluginapi.Client, log utils.Logger, conf config.Service, aws upaws.Client, s3AssetBucket string, store *store.Service, mutex *cluster.Mutex, httpOut httpout.Service) *Proxy {
+func NewService(mm *pluginapi.Client, log utils.Logger, conf config.Service, store *store.Service, mutex *cluster.Mutex, httpOut httpout.Service) *Proxy {
 	return &Proxy{
 		builtinUpstreams: map[apps.AppID]upstream.Upstream{},
 		mm:               mm,
 		log:              log,
 		conf:             conf,
 		store:            store,
-		aws:              aws,
-		s3AssetBucket:    s3AssetBucket,
 		callOnceMutex:    mutex,
 		httpOut:          httpOut,
 	}
 }
 
+func (p *Proxy) Configure(conf config.Config) error {
+	newUpstream := func(appType apps.AppType, makeUpstream func() (upstream.Upstream, error)) {
+		if isAppTypeSupported(conf, appType) == nil {
```

Comment:
Can we log the error `isAPpTypeSupported` returns?

Reply:
2/5 We shouldn't - we'd be routinely logging things like "App type plugin is not supported on Mattermost Cloud"?